### PR TITLE
properly curry ifElse; fixes #939

### LINKED ISSUE #939: function returned by R.ifElse has incorrect length
Clearly `R.ifElse(R.gt, R.subtract, R.add)` evaluates to a binary function:

```javascript
> R.ifElse(R.gt, R.subtract, R.add)(7, 2)
5
> R.ifElse(R.gt, R.subtract, R.add)(2, 7)
9
```

But it does not report its length correctly:

```javascript
> R.ifElse(R.gt, R.subtract, R.add).length
0
```

Also, partial application does not work as expected:

```javascript
> R.ifElse(R.gt, R.subtract, R.add)(2)(7)
-5
```

### PR DESCRIPTION
**Pull request by buzzdecafe**

### CODE REVIEW

File: test/ifElse.js [hunk 1]
```diff
@@ -42,10 +42,25 @@ describe('ifElse', function() {
         R.ifElse(v, identity, onFalse)(123, 'abc');
     });
 
+    it('returns a function whose arity equals the max of the three arguments to `ifElse`', function() {
+        function a0() { return 0; }
+        function a1(x) { return x; }
+        function a2(x, y) { return x + y; }
+
+        var fn = R.ifElse(a1, a2, a0);
+        assert.strictEqual(fn.length, a2.length);
```
Comment (davidchambers) [inline test/ifElse.js]: Let's test all the permutations:

```javascript
assert.strictEqual(R.ifElse(a0, a1, a2).length, 2);
assert.strictEqual(R.ifElse(a0, a2, a1).length, 2);
assert.strictEqual(R.ifElse(a1, a0, a2).length, 2);
assert.strictEqual(R.ifElse(a1, a2, a0).length, 2);
assert.strictEqual(R.ifElse(a2, a0, a1).length, 2);
assert.strictEqual(R.ifElse(a2, a1, a0).length, 2);
```

File: test/ifElse.js [hunk 2]
```diff
@@ -42,10 +42,25 @@ describe('ifElse', function() {
         R.ifElse(v, identity, onFalse)(123, 'abc');
     });
 
+    it('returns a function whose arity equals the max of the three arguments to `ifElse`', function() {
```
Comment (davidchambers) [inline test/ifElse.js]: s/max/max arity/

File: test/ifElse.js [hunk 3]
```diff
@@ -42,10 +42,25 @@ describe('ifElse', function() {
...
         var v = function(a) { return typeof a === 'number'; };
         var ifIsNumber = R.ifElse(v);
         assert.strictEqual(ifIsNumber(t, identity)(15), 16);
         assert.strictEqual(ifIsNumber(t, identity)('hello'), 'hello');
+
+        var fn = R.ifElse(R.gt, R.subtract, R.add);
+        assert.strictEqual(fn(2)(7), 9);
+        assert.strictEqual(fn(2, 7), 9);
+        assert.strictEqual(fn(7)(2), 5);
+        assert.strictEqual(fn(7, 2), 5);
```
Comment (davidchambers) [inline test/ifElse.js]: :thumbsup:

File: src/ifElse.js [hunk 4]
```diff
@@ -23,7 +24,8 @@ var _curry3 = require('./internal/_curry3');
  *      flattenArrays([[[10], 123], [8, [10]], "hello"]); //=> [[10, 123], [8, 10], "hello"]
  */
 module.exports = _curry3(function ifElse(condition, onTrue, onFalse) {
-    return function _ifElse() {
+    var len = Math.max(condition.length, onTrue.length, onFalse.length);
```
Comment (davidchambers) [inline src/ifElse.js]: You could remove this variable.

**Comment by CrossEye** [comment]
:herb:

**Pull request merged**